app/app.go

```diff
@@ -26,6 +27,9 @@ type App struct {
 	// Main (uncached) state
 	ms sdk.CommitMultiStore
 
+	// mapping from strings to keys to allow queries
+	storeKeys map[string]sdk.SubstoreKey
```

Comment:
store/rootmultistore.go/rootMultiStore should have a field `storeKeysByName map[string]sdk.StoreKey` (SubstoreKey became StoreKey).  rootMultiStore.MountStoreWithDB() should check this new map after checking for key == nil... otherwise the other maps in rootMultiStore look OK.

Reply:
Okay, but how does the app read this?

If the app can get all the store keys (or look up by string), then so can any other handler that is passed the multistore. The only way I could think of to give this info to the app but not the handlers is to pass this info outside the multistore.

How do we handle access to `storeKeysByName`?